src/javascript/app/Modules/Reports/Helpers/shortcode.js

```diff
@@ -1,26 +1,28 @@
 const Shortcode = (() => {
-    let info_from_shortcode = {};
+    const info_from_shortcode = {
+        category  : '',
+        underlying: '',
+        atm       : '',
+    };
 
     const extractInfoFromShortcode = (shortcode) => {
-        const pattern = new RegExp('^([A-Z]+)_((OTC_[A-Z0-9]+)|R_[\\d]{2,3}|[A-Z]+)_([\\d.\\d]+)_(\\d+\\w)_(\\d+\\w)_(S[\\d.\\d]+P|(?:\\d+))_(\\d+)'); // Used to get market name from shortcode
+        const pattern = new RegExp('^([A-Z]+)_((OTC_[A-Z0-9]+)|R_[\\d]{2,3}|[A-Z]+)'); // Used to get market name from shortcode
         const extracted = pattern.exec(shortcode);
         if (extracted !== null) {
             info_from_shortcode.category   = extracted[1].toLowerCase();
             info_from_shortcode.underlying = extracted[2];
-            info_from_shortcode.atm        = extracted[7];
+
+            if (/CALL|PUT/i.test(info_from_shortcode.category)) {
```

Comment:
it should have `^` to make sure it's at the beginning of the string, so `/^(CALL|PUT)/`